driver/tab_completion.py

```diff
@@ -0,0 +1,122 @@
+import argparse
+import importlib.util
+from pathlib import Path
+import subprocess
+import sys
+
+from . import util
+
+try:
+    import argcomplete
+    HAS_ARGCOMPLETE = True
+except ImportError:
+    HAS_ARGCOMPLETE = False
+
+
+def complete_build_arg(prefix, parsed_args, **kwargs):
+    if parsed_args.debug:
+        argcomplete.warn("The option --debug is an alias for --build=debug. Do no specify both --debug and --build.")
+        exit(1)
+
+    builds_folder = Path(util.REPO_ROOT_DIR) / "builds"
+    if not builds_folder.exists():
+        argcomplete.warn("No build exists.")
+        exit(1)
+    return [p.name for p in builds_folder.iterdir() if p.is_dir()]
+
+
+def complete_planner_args(prefix, parsed_args, **kwargs):
+    if parsed_args.build and parsed_args.debug:
+        argcomplete.warn("The option --debug is an alias for --build=debug. Do no specify both --debug and --build.")
+        exit(1)
+
+    build = parsed_args.build
+    if not build:
+        if parsed_args.debug:
+            build = "debug"
+        else:
+            build = "release"
+
+    # Get some information from planner_args before it is deleted in split_planner_args().
+    planner_args = parsed_args.planner_args
+    num_planner_args = len(planner_args)
+    mode_switches = ["--translate-options", "--search-options"]
+    last_option_was_mode_switch = planner_args and (planner_args[-1] in mode_switches)
+    double_dash_in_options = "--" in planner_args
+
+    current_mode = util.split_planner_args(parsed_args)
+    num_filenames = len(parsed_args.filenames)
+    has_only_filename_options = (num_filenames == num_planner_args)
+    has_only_filename_or_double_dash_options = (num_filenames + int(double_dash_in_options) == num_planner_args)
+    can_use_double_dash = (1 <= num_planner_args <= 2) and has_only_filename_or_double_dash_options
+
+    completions = []
+
+    if can_use_double_dash:
+        completions.append("--")
+
+    if parsed_args.filenames or double_dash_in_options:
+        bin_dir = Path(util.REPO_ROOT_DIR) / "builds" / build / "bin"
```

Comment:
It's not good that information about where certain files are located is now repeated in many places in the code. We already have a central place for this and shouldn't reimplement this logic.

Reply:
Yes, but it probably makes sense to unify it after merging the pull request #228 which adds `pathlib` support.

Reply:
This place now uses the existing function. Malte mentioned "many places". Are there other places than this that we should fix?

Reply:
In the current version, I see only one other somewhat related place, although it's not quite the same thing. I'll add a comment there.